bitsandbytes/diagnostics/cuda.py

```diff
@@ -105,37 +108,58 @@ def find_cudart_libraries() -> Iterator[Path]:
-    )
+    if not HIP_ENVIRONMENT:
+        print(
+            f"PyTorch settings found: CUDA_VERSION={cuda_specs.cuda_version_string}, "
+            f"Highest Compute Capability: {cuda_specs.highest_compute_capability}.",
+        )
+    else:
+        print(f"PyTorch settings found: ROCM_VERSION={cuda_specs.cuda_version_string}")
 
     binary_path = get_cuda_bnb_library_path(cuda_specs)
     if not binary_path.exists():
-        print_dedented(
-            f"""
-        Library not found: {binary_path}. Maybe you need to compile it from source?
-        If you compiled from source, try again with `make CUDA_VERSION=DETECTED_CUDA_VERSION`,
-        for example, `make CUDA_VERSION=113`.
-
-        The CUDA version for the compile might depend on your conda install, if using conda.
-        Inspect CUDA version via `conda list | grep cuda`.
-        """,
-        )
-
-    cuda_major, cuda_minor = cuda_specs.cuda_version_tuple
-    if cuda_major < 11:
-        print_dedented(
-            """
-            WARNING: CUDA versions lower than 11 are currently not supported for LLM.int8().
-            You will be only to use 8-bit optimizers and quantization routines!
+        if not HIP_ENVIRONMENT:
+            print_dedented(
+                f"""
+            Library not found: {binary_path}. Maybe you need to compile it from source?
+            If you compiled from source, try again with `make CUDA_VERSION=DETECTED_CUDA_VERSION`,
+            for example, `make CUDA_VERSION=113`.
+
+            The CUDA version for the compile might depend on your conda install, if using conda.
+            Inspect CUDA version via `conda list | grep cuda`.
             """,
-        )
+            )
+        else:
+            print_dedented(
+                f"""
+            Library not found: {binary_path}.
+            Maybe you need to compile it from source? If you compiled from source, check that ROCM_VERSION
+            in PyTorch Settings matches your ROCM install. If not, reinstall PyTorch for your ROCm version
+            and rebuild bitsandbytes.
+            """,
+            )
 
-    print(f"To manually override the PyTorch CUDA version please see: {NONPYTORCH_DOC_URL}")
+    cuda_major, cuda_minor = cuda_specs.cuda_version_tuple
+    if not HIP_ENVIRONMENT:
+        if cuda_major < 11:
+            print_dedented(
+                """
+                WARNING: CUDA versions lower than 11 are currently not supported for LLM.int8().
+                You will be only to use 8-bit optimizers and quantization routines!
+                """,
+            )
+
+        print(f"To manually override the PyTorch CUDA version please see: {NONPYTORCH_DOC_URL}")
+    else:
+        if (cuda_major, cuda_minor) < (6, 1):
+            print_dedented(
+                """
+                WARNING: bitandbytes is fully supported only from ROCm 6.1.
+                """,
+            )
```

Comment:
Instead of these nested branches, it would probably be cleaner to simply split this out to two functions, `_print_nvidia_cuda_diagnostics()` and `_print_hip_diagnostics()`.